### PR TITLE
Extended character parser for quarters

### LINKED ISSUE #107: Custom yearquarter() character parser
As `anytime` cannot parse `"2004 Q2"` or `"Q2 2004"`, can we add a simple parser for these formats? If the input format does not match this, it could fall back to anytime.

cc @robjhyndman

### PR DESCRIPTION
**Pull request by mitchelloharawild**
Resolves #107
cc @robjhyndman

### CODE REVIEW

File: R/period.R [hunk 1]
```diff
@@ -402,8 +402,18 @@ yearquarter.Date <- yearquarter.POSIXt
 yearquarter.character <- function(x) {
   if (is_empty(x)) return(as_yearquarter(x))
 
-  anytime::assertDate(x)
-  as_yearquarter(anytime::anydate(x))
+  qtr_regex <- "(?i)(q|qtr|quarter)(?-i) *"
+  if(all(grepl(paste0("\\d{4} *", qtr_regex, "\\d|", qtr_regex, "\\d *\\d{4}"), x, perl = TRUE))){
```
Comment (earowang) [inline R/period.R]: 1. Can you put space around the `if-else` statement?
2. In the condition, can you just do `grepl(qtr_regex, x, perl = TRUE)` without checking other things?

Comment (earowang) [inline R/period.R]: Can you refactor the code to extract digits only without using `qtr_regrex` for quarter and year?

File: tests/testthat/test-index-fun.R [hunk 2]
```diff
@@ -81,6 +81,9 @@ test_that("some S3 methods for yearweek, yearmonth & yearquarter", {
   expect_is(unique(y2), "yearquarter")
   expect_identical(yearquarter(y), y)
   expect_is(y[1:2], "yearquarter")
+  expect_equal(yearquarter("2010 Q1"), y[1])
```
Comment (earowang) [inline tests/testthat/test-index-fun.R]: Can you make a new test block to include the tests, and refer to the issue in the `test_that(..., {})`?

**Review by earowang — CHANGES_REQUESTED** [review]

**Review by earowang — CHANGES_REQUESTED** [review]
can you please create a new branch other than master?

File: R/period.R (hunk 1 above)
Comment (mitchelloharawild) [reply R/period.R]: 1. Fixed
2. Why? It's safer to match the whole expression, rather than accept an incomplete expression.

Comment (mitchelloharawild) [reply R/period.R]: You can refactor it - using `qtr_regex` for both quarter and year is safer than just matching digits.

File: tests/testthat/test-index-fun.R (hunk 2 above)
Comment (mitchelloharawild) [reply tests/testthat/test-index-fun.R]: Fixed.

**Comment by mitchelloharawild** [comment]
I can use branches for the next PR - to change branches would require a new PR.

**Pull request closed without merging**